tests/bosonic_files/test_backend_bosoniccircuit.py

```diff
@@ -0,0 +1,406 @@
+# Copyright 2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+r"""
+Unit tests for backends.bosonicbackend.bosoniccircuit.py.
+"""
+
+import numpy as np
+import strawberryfields.backends.bosonicbackend.bosoniccircuit as circuit
+import pytest
+
+pytestmark = pytest.mark.bosonic
+
+NUM_WEIGHTS_VALS = [1, 2, 5]
+NUM_MODES_VALS = [1, 3, 4]
+NUM_NEW_WEIGHTS_VALS = [1, 2, 3]
+R_VALS = np.array([-1.0, -1 / 3, 1 / 2, 1.2])
+PHI_VALS = np.array([-1.2, -1 / 5, 0.6, 1.3])
+NBAR_VALS = np.array([0, 1, 2.5])
+NUM_SHOTS_VALS = np.array([1, 2, 10])
+R_ANC_VALS = np.array([0.1, 0.8, 1.2, 5])
+ETA_ANC_VALS = np.array([0.1, 0.4, 0.9, 1])
+
+
+class TestBosonicCircuit:
```

Comment:
Suggest aligning this with the name of the class (currently `BosonicModes`, but perhaps that will change?)

Reply:
Done.